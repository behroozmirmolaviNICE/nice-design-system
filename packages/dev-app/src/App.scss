@import "~@nice-digital/design-system/scss/nice-design-system";

@import "~@nice-digital/nds-forms/scss/forms";

main {
  background: white;
  box-shadow: 0 0 10px rgba(0, 0, 0, .3);
  margin: 3vw;
  padding: 3vw;
  position: relative;
}
