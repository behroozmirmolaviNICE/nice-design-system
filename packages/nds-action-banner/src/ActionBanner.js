// @flow
import React, { Component } from "react";
import "../scss/action-banner.scss";

type ActionBannerProps = {
	title: string,
	modifier: "subtle",
	children: *,
	cta: Element,
	closeable: boolean
};

const ActionBanner = (props: ActionBannerProps) => {

	let classes = ["action-banner"];

	if(props.modifier)
		classes.push(`action-banner--${props.modifier}`);

	if(props.closeable)
		classes.push("action-banner--closeable");

	return <section className={classes.join(" ")}>
		<div className="action-banner__container">
			<div className="action-banner__inner">
				<div className="action-banner__text">
					<h2 className="action-banner__title">
						{props.title}
					</h2>
					<p className="action-banner__intro">
						{props.children}
					</p>
				</div>
				<div className="action-banner__actions">
					{props.cta}
				</div>
				{props.closeable &&
					<button type="button" className="action-banner__close">
						<span className="icon icon--remove" ara-hidden="true"></span>
						<span className="visually-hidden">Close</span>
					</button>
				}
			</div>
		</div>
	</section>;
};

export default ActionBanner;
