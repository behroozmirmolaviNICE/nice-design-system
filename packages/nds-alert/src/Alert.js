// @flow
import React, {Component} from "react";
import classnames from "classnames";
import "../scss/alert.scss";

type AlertProps = {
	children: React.Node,
	type: string,
};

class Alert extends Component<AlertProps> {
	render(){
		const { children } = this.props;
		let type = this.props.type || "info";
		const classNames = classnames({
			"alert": true,
			[`alert--${type}`]: true
		});
		return (
			<div className={classNames}>
				{children}
			</div>
		);
	}
}

export default Alert;
