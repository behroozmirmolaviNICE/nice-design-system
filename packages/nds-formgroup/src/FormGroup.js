import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "./../scss/formgroup.scss";

export const FormGroup = props => {
	const { groupError, inline, legend, children, hint, name, ...rest } = props;

	const clonedChildren = React.Children.map(children, child => {
		return React.cloneElement(child, {
			name,
			inline,
			...rest
		});
	});

	const classes = classnames({
		formgroup: true,
		"formgroup--no-legend": legend ? false : true
	});

	return (
		<fieldset className={classes}>
			{legend && <legend className="formgroup__legend">{legend}</legend>}
			{groupError && <p className="formgroup__error-message">{groupError}</p>}
			{hint && <p className="formgroup__hint">{hint}</p>}
			{clonedChildren}
		</fieldset>
	);
};

FormGroup.propTypes = {
	children: PropTypes.node.isRequired,
	legend: PropTypes.node,
	name: PropTypes.string,
	hint: PropTypes.string,
	inline: PropTypes.bool,
	groupError: PropTypes.string
};
