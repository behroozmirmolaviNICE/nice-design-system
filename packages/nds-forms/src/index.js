import Radio from "./Radio";
import RadioGroup from "./RadioGroup";
import Textarea from "./Textarea";
import Input from "./Input";
import Fieldset from "./Fieldset";

export { Radio, RadioGroup, Textarea, Input, Fieldset };
