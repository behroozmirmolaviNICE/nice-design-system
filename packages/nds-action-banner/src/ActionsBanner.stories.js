import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs } from "@storybook/addon-knobs";

import ActionBanner from "./ActionBanner";

storiesOf("Action banner", module)
	.addDecorator(withKnobs)
	.add("Primary", () => <ActionBanner />);
