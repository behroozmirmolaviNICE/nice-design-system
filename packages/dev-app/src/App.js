import React, { Component } from "react";
import { Radio, RadioGroup, Textarea, Input, Fieldset, Checkbox } from "@nice-digital/nds-forms";

import "./App.scss";

class App extends Component {
	render() {
		return (
			<div>

				<Fieldset legend="Checkboxes">
					<Checkbox />
				</Fieldset>

				<RadioGroup
					legend="Radio buttons (stacked, default)"
					group="group-1"
					hint="This is some hint text you can read">
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..? Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ea eaque explicabo, facere harum necessitatibus nulla optio pariatur rem! Aut blanditiis dolores expedita minima mollitia nostrum rem! Accusamus aliquid, distinctio facilis harum illum magni nemo pariatur provident sequi vel, voluptates voluptatum!"/>
				</RadioGroup>

				<RadioGroup legend="Radio buttons inline" group="group-2" hint="This is some hint text you can read" inline>
					<Radio value="yes" label="Yes please!"/>
					<Radio value="no" label="No thanks!"/>
					<Radio value="maybe" label="Well maybe..?"/>
				</RadioGroup>

				<Fieldset legend="Textarea">
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						value="Default value for textarea"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						hint="This is some hint text you can read"
						name="myCommentBox"/>
					<Textarea
						unique="textarea-1"
						label="Who do you think you are?"
						name="mySecondCommentBox"/>
				</Fieldset>

				<Fieldset legend="Inputs">
					<Input
						name="full-name"
						type="text"
						label="Full name"/>
					<Input
						name="email"
						type="email"
						label="Your email address"/>
					<Input
						name="password"
						type="password"
						hint="Your password must contain 4 special characters, 6 numbers, 12 letters and an emoji"
						label="Password"/>
				</Fieldset>

			</div>
		);
	}
}

export default App;
