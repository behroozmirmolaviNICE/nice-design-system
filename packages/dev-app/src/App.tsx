import React, { Component } from "react";
// import Faker from "faker";

import { Checkbox } from "@nice-digital/nds-checkbox";
import "./App.scss";

class App extends Component {
	render() {
		return (
			<main>
				<Checkbox />
			</main>
		);
	}
}

export default App;
