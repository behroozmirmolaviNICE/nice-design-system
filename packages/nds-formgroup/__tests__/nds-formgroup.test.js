"use strict";

import React from "react";
import { shallow, mount } from "enzyme";
import { FormGroup } from "./../src/FormGroup";

describe("FormGroup", () => {
	it("should render without crashing", () => {
		const wrapper = shallow(<FormGroup />);
		expect(wrapper).toHaveLength(1);
	});
});
