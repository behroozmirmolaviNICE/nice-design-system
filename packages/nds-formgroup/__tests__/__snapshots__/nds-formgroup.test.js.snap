// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup should match the snapshot when all possible props are passed 1`] = `
<fieldset
  className="formgroup"
>
  <legend
    className="formgroup__legend"
  >
    This is the legend text
  </legend>
  <p
    className="formgroup__error-message"
  >
    This is the error text
  </p>
  <p
    className="formgroup__hint"
  >
    This is the hint text
  </p>
  <span
    inline={false}
    name="group-name"
  >
    Render
  </span>
</fieldset>
`;

exports[`FormGroup should match the snapshot when no props are passed 1`] = `
<fieldset
  className="formgroup formgroup--no-legend"
>
  <span>
    Render
  </span>
</fieldset>
`;
